### ISSUE TITLE
Add tests

### ISSUE THREAD

**Issue by ConnorMac** [rehive/rehive-python#14]
We should ideally have tests that we can run before compiling and releasing a new version into the wild.

EDIT (by @OdinsPlasmaRifle ) The goal of this issue is to add a basic unit testing framework for the SDK.

**Comment by ConnorMac** [comment]
A note on this. It could be worth having a `test mode` flag(ENV variable) which stubs out all calls the SDK makes so any service can easily write tests without manually stubbing all the calls it needs.

**Comment by OdinsPlasmaRifle** [comment]
This sounds like a really awesome idea, would allow users to build software without having to make a Rehive account.

**Comment by ConnorMac** [comment]
Indeed. That would be a pretty cool use case. The primary just being a great way to save a lot of grunt work in the future.

It does mean we have a whole lot more to maintain in the SDK though. Since the stubbed calls would need to be kept up to date(but they will need to be updated on every service that has tests anyway). We could consider writing something that automates making a company and then runs through all endpoints getting the response and saving the diff. This could be useful for docs as well.

**Comment by OdinsPlasmaRifle** [comment]
Could this perhaps work by using the Swagger schema to generate stubs in some way? 🤔
